nilearn/_utils/data_gen.py

```diff
@@ -274,6 +274,8 @@ def generate_fake_fmri(shape=(10, 11, 12), length=17, kind="noise",
     width = [s // 2 for s in shape]
     shift = [s // 4 for s in shape]
 
+    if rand_gen is None:
```

Comment:
I think you should call `check_random_state` without any condition since the user can pass either None, an int, or a random state as input, and the scikit-learn function is clever enough to handle this.

Reply:
Sure. I will update it

Reply:
I have fixed this in the latest commit